core/lib/zksync_core/src/metadata_calculator/api.rs

```diff
@@ -0,0 +1,215 @@
+//! Primitive Merkle tree API used internally to fetch proofs.
+
+use anyhow::Context as _;
+use async_trait::async_trait;
+use axum::{
```

Comment:
I've implemented HTTP API using `axum` since it seems reasonably lightweight. We could try using something more performant (gRPC with `prost` / `tonic`?), but since it's not used in the codebase and this is intended to be internal API, it looks like an overkill.